### PR TITLE
Fix TravisCI by installing latest npm

### PR DESCRIPTION
**Pull request by hawkrives**
Closes https://github.com/gatsbyjs/gatsby/issues/984

I took a stab at cleaning up the 1.0 `.travis.yml`, and (if I install npm@5 first) everything passes properly.

I re-enabled mac builds, and put node@4 back in, too. (Here's an [example build](https://travis-ci.org/hawkrives/gatsby/builds/237372618), along with the [config](https://github.com/hawkrives/gatsby/blob/1.0/.travis.yml).)

### CODE REVIEW

**Comment by hawkrives** [comment]
Should I upgrade appveyor, as well?

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit ba52bb96ea9fd6226ac78698b37f5255edf222d9

https://app.netlify.com/sites/image-processing/deploys/592dd565424ef204b9332d77

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit ba52bb96ea9fd6226ac78698b37f5255edf222d9

https://app.netlify.com/sites/gatsbygram/deploys/592dd565424ef204b9332d68

**Comment by KyleAMathews** [comment]
>appveyor

ooo that'd be great! @jbolda has been working on it at https://github.com/gatsbyjs/gatsby/pull/1049

I tried too for a little while but couldn't see why the lerna install is failing there.

Re: npm5 — we use yarn for installing stuff so I'm not sure why installing npm5 or not matters? Perhaps just drop installing that to speed things up a bit?

**Comment by jbolda** [comment]
I've always had issues with Lerna on Windows... If you can figure it out, @hawkrives, go for it. @kyleamathews I did email them, but haven't heard back.

**Comment by jbolda** [comment]
@hawkrives any thoughts to building the examples sites as well?

**Comment by hawkrives** [comment]
@jbolda Sure, I can take a look at it. 

@KyleAMathews The npm5 install has been averaging ~15 seconds… I think the build failed because it's not marked as "latest" yet. I'm going to try forcing v5; assuming that _works_, … well, ideally there wouldn't be any difference between npm and yarn.

**Comment by hawkrives** [comment]
@jbolda Where do the example sites live? Are they the "-starter-" repos under @gatsbyjs?

**Comment by hawkrives** [comment]
… I found the examples. I was looking under `master` again.

**Comment by hawkrives** [comment]
For the examples: are you OK with only testing them on one platform/node version … or do you want them all tested on all six?

**Comment by hawkrives** [comment]
Ah, ok. This build might be going a bit overboard then: https://travis-ci.org/hawkrives/gatsby/builds/237779366

**Comment by KyleAMathews** [comment]
Haha yes... quite possibly ;-)

TravisCI also puts a limit on the number of builds that can run in parallel as you can see so this would mean *very* slow test times.

**Comment by KyleAMathews** [comment]
And I got to move gatsbybot to a different email address :-|

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,25 @@
...
-      - ubuntu-toolchain-r-test
-    packages:
-      - g++-4.8
+  - "8"
+
+os: [linux, osx]
+dist: trusty
+sudo: required
+
+# by caching the npm local cache, instead of ./node_modules,
```
Comment (KyleAMathews) [inline .travis.yml]: Not sure I understand what this is for?

Comment (hawkrives) [inline .travis.yml]: I might not even need it with npm@5. I'll check.

**Comment by KyleAMathews** [comment]
@hawkrives this is great! Thanks for cleaning things up! Merging now.

**Comment by hawkrives** [comment]
✅
I'll probably poke at Appveyor some this weekend

**Comment by KyleAMathews** [comment]
Ok, I think I remember now why I disabled testing on macs. Running tests on Travis are now taking ~30-40 minutes. Travis only runs so many tests per project at a time and with the extra versions of node and mac, we've gone from 3-5 minute turnaround on tests to 10x that.

I think I'll trim things down to only run node 8 on mac and only 6/8 on linux. Our babel config is set to compile code to be compatible w/ node 4 so we should be fine-ish there. Moving fast is better than never breaking things.

**Comment by KyleAMathews** [comment]
Hmm you can't actually just do node 8 on macs. I'll just drop testing on macs. I develop Gatsby on a mac as do many other people so the minor differences between mac/linux should be accounted for by manual testing.

**Comment by hawkrives** [comment]
I have node 8 working on macs?

**Comment by KyleAMathews** [comment]
@hawkrives were tests not working on node 8 on macs?

**Comment by hawkrives** [comment]
@KyleAMathews AFAICT, the tests were running and passing in node8 on macOS in Travis when you merged this? or am I misreading something (entirely possible)?

**Comment by KyleAMathews** [comment]
Yes they were/are. I thought you were saying earlier that they weren't but [you'd done something to get them working](https://github.com/gatsbyjs/gatsby/pull/1066#issuecomment-305883260).

**Comment by hawkrives** [comment]
Ah, yeah, sorry. Poor choice of wording/punctuation on my part.

**Comment by jlengstorf** [comment]
Hiya @hawkrives! 👋

This is definitely late, but on behalf of the entire Gatsby community, I wanted to say thank you for being here.

Gatsby is built by awesome people like you. Let us say “thanks” in two ways:

1.  **We’d like to send you some Gatsby swag.** As a token of our appreciation, you can go to the [Gatsby Swag Store][store] and log in with your GitHub account to get a coupon code good for one free piece of swag. (We’ve got t-shirts and hats, plus some socks that are really razzing our berries right now.)
2.  **If you’re not already part of it, we just invited you to join the Gatsby organization on GitHub.** This will add you to our team of maintainers. You’ll receive an email shortly asking you to confirm. By joining the team, you’ll be able to label issues, review pull requests, and merge approved pull requests.

If you have questions, please don’t hesitate to reach out to us: tweet at [@gatsbyjs][twitter] and we’ll come a-runnin’.

Thanks again! 💪💜

[store]: https://store.gatsbyjs.org
[twitter]: https://twitter.com/gatsbyjs

**Pull request merged**